src/hydra_zen/wrapper/_implementations.py

```diff
@@ -629,3 +642,333 @@ def wrap(f: Callable[P, R]) -> Zen[P, R]:
+                deferred_to_config=self._deferred_to_config,
+                deferred_hydra_store=self._deferred_store,
+                overwrite_ok=self._overwrite_ok,
+            )
+            _s._defaults = self._defaults.copy()
+
+            # Important: mirror internal state *by reference* to ensure `_s` and
+            # `self` remain in sync
+            _s._internal_repo = self._internal_repo
+            _s._queue = self._queue
+
+            new_defaults: _Defaults = {k: kw[k] for k in _DEFAULT_KEYS if k in kw}  # type: ignore
+
+            new_defaults["__kw"] = {
+                **_s._defaults["__kw"],
+                **{k: kw[k] for k in set(kw) - _DEFAULT_KEYS},
+            }
+            _s._defaults.update(new_defaults)
+            return _s
+        else:
+            to_config = kw.get("to_config", self._defaults["to_config"])
+            name = kw.get("name", self._defaults["name"])
+            group = kw.get("group", self._defaults["group"])
+            package = kw.get("package", self._defaults["package"])
+            provider = kw.get("provider", self._defaults["provider"])
+
+            _name: NodeName = name(__target) if callable(name) else name
+            if not isinstance(_name, str):  # type: ignore
+                raise TypeError(f"`name` must be a string, got {_name}")
+            del name
+
+            _group: GroupName = group(__target) if callable(group) else group
+            if _group is not None and not isinstance(_group, str):  # type: ignore
+                raise TypeError(f"`group` must be a string or None, got {_group}")
+            del group
+
+            _pkg = package(__target) if callable(package) else package
+            if _pkg is not None and not isinstance(_pkg, str):
+                raise TypeError(f"`package` must be a string or None, got {_pkg}")
+            del package
+
+            merged_kw = {
+                **self._defaults["__kw"],
+                **{k: kw[k] for k in set(kw) - _DEFAULT_KEYS},
+            }
+
+            if self._deferred_to_config:
+
+                def deferred_to_config():
+                    return to_config(__target, **merged_kw)  # noqa: E731
+
+                node = deferred_to_config
+            else:
+                node = to_config(__target, **merged_kw)
+
+            entry = _Entry(
+                name=_name,
+                group=_group,
+                package=_pkg,
+                provider=provider,
+                node=node,
+            )
+
+            if not self._overwrite_ok and (_group, _name) in self._internal_repo:
+                raise ValueError(
+                    f"(name={entry['name']} group={entry['group']}): "
+                    f"Hydra config store entry already exists. Specify "
+                    f"`overwrite_ok=True` to enable replacing config store entries"
+                )
+            self._internal_repo[_group, _name] = entry
+            self._queue.append(entry)
+
```

Comment:
I think `self._internal_repo` should be keyed on the triplet `(_group, _name, _package)` rather than on the pair `(_group, _name)`, and the `ValueError` should be raised if `(_group, _name, _package) in self._internal_repo`.

Reply:
Ok nevermind, disregard my previous comment.
I see this wouldn't work, as Hydra's `ConfigStore` is keyed on `(group, name)`.